src/test/crimson/test_alien_echo.cc

```diff
@@ -93,35 +96,58 @@ struct Client {
 } // namespace seastar_pingpong
 
 class SeastarContext {
-  seastar::file_desc begin_fd;
-  seastar::readable_eventfd on_end;
+  int begin_fd;
+  seastar::file_desc on_end;
 
 public:
   SeastarContext()
-    : begin_fd{seastar::file_desc::eventfd(0, 0)}
+    : begin_fd{eventfd(0, 0)},
+      on_end{seastar::file_desc::eventfd(0, 0)}
   {}
 
   template<class Func>
   std::thread with_seastar(Func&& func) {
-    return std::thread{[this, func = std::forward<Func>(func)] {
+    return std::thread([this, on_end = on_end.get(),
+		        func = std::forward<Func>(func)] {
         // alien: are you ready?
         wait_for_seastar();
         // alien: could you help me apply(func)?
         func();
         // alien: i've sent my request. have you replied it?
         // wait_for_seastar();
         // alien: you are free to go!
-        on_end.write_side().signal(1);
-      }};
+        ::eventfd_write(on_end, 1);
+      });
   }
 
   void run(seastar::app_template& app, int argc, char** argv) {
     app.run(argc, argv, [this] {
-      return seastar::now().then([this] {
-        return set_seastar_ready();
+      return crimson::common::sharded_conf().start(EntityName{}, string_view{"ceph"})
+      .then([] {
+        std::vector<const char*> args;
+        std::string cluster;
+        std::string conf_file_list;
+        auto init_params = ceph_argparse_early_args(args,
+                                                    CEPH_ENTITY_TYPE_CLIENT,
+                                                    &cluster,
+                                                    &conf_file_list);
+        auto& conf = crimson::common::local_conf();
+        conf->name = init_params.name;
+        conf->cluster = cluster;
```

Comment:
Could you move them before `crimson::common::sharded_conf().start(...)`, so no need to provide dummy names and have to correct them later.

Ditto `test_messenger.cc`

Reply:
done.